src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -110,13 +56,13 @@ public static JsonObject verifyAndExtractBody(Response response) {
   /**
    * If response http code is {@code 404}, empty {@link JsonObject} is returned. Otherwise verifies if response is successful and extracts body
    * In case there was failed attempt to delete PO or particular PO line, the sub-objects might be already partially deleted.
-   * This check allows user to retrieve order/line again and retry DELETE operation
+   * This check allows user to retrieve order/line again and retry DELETE operation if required
    *
    * @param response response to verify
    * @return empty {@link JsonObject} if response http code is {@code 404}, otherwise calls {@link #verifyAndExtractBody(Response)}
    */
-  private static JsonObject verifyAndExtractBodyIfFound(Response response) {
-    if (response.getCode() == 404) {
+  private static JsonObject verifyAndExtractBody(HttpMethod operation, Response response) {
+    if (response.getCode() == 404 && operation == HttpMethod.GET) {
```

Comment:
See my comment in `operateOnSubObj(operation, response)`

Reply:
The method is removed and the logic moved to `operateOnSubObj`